### PR TITLE
Added guard against multiple connects

### PR DESCRIPTION
**Pull request by mikaelkindborg**
Added function to Bleat Classic API to check if a device is connected. Also added guard against multiple connects to the Evothings adapter.

### CODE REVIEW

**Comment by thegecko** [comment]
I can't see the changes in `adapter.evothings.js` It looks like every line has changed.

File: dist/api.classic.js [hunk 1]
```diff
@@ -111,7 +113,11 @@
             executeFn(disconnectFn)();
         }.bind(this), raiseError(errorFn, "connect error"));
     };
+    Device.prototype.isConnected = function() {
+        return this.connected;
```
Comment (thegecko) [inline dist/api.classic.js]: this should be initiated in the device constructor.

Comment (thegecko) [inline dist/api.classic.js]: In fact, if it is initiated, you don't need a getter function and can just use the device property `device.connected`

File: dist/api.classic.js [hunk 2]
```diff
@@ -111,7 +113,11 @@
...
+    };
     Device.prototype.disconnect = function(errorFn) {
+        this.connected = false;
```
Comment (thegecko) [inline dist/api.classic.js]: Perhaps the connected state should be exposed by each adapter to keep it atomic? Is the property kept in sync with reality i.e. if the device powers off?

File: dist/api.classic.js [hunk 3]
```diff
@@ -111,7 +113,11 @@
...
         adapter.disconnect(this._handle, raiseError(errorFn, "disconnect error"));
```
Comment (thegecko) [inline dist/api.classic.js]: If `adapter.disconnect` fails, the connected property will be wrong.

File: dist/api.classic.js (hunk 2 above)
Comment (mikaelkindborg) [inline dist/api.classic.js]: @thegecko Sounds like a good idea to let the adapter keep track of the connected state. Will look into this!

**Comment by mikaelkindborg** [comment]
@thegecko Hi thanks a lot for reviewing the code. Will look into improving the connected handling and keep the white space cleanup separate so it is easier to see changes. Sounds like a good idea to make knowing the connected state the responsibility of the adapter.

**Comment by mikaelkindborg** [comment]
@thegecko Hi, I cleaned up the pull request, keeping only the already-connected check in the evothings adapter.

**Comment by thegecko** [comment]
@mikaelkindborg Sorry about the delay on this merge, are you still waiting for it?

**Comment by mikaelkindborg** [comment]
@thegecko Yes, do you think it is ok now? I removed much of the code that was problematic. 

Been working on improving the Evothings BLE plugin during the the summer. In the work I do, focus is now on Web Bluetooth rather than Bleat Classic. I am happy to help support development of Bleat and the Web Bluetooth library.

**Pull request merged**